### PR TITLE
Add capabilities attributes/properties for MusicProviders

### PR DESCRIPTION
**Pull request by marcelveldt**
This comes in handy when/if a provider does not support all features or is in the process of being added

### CODE REVIEW

**Comment by MarvinSchenkel** [comment]
Can't we turn this into an opt-in thing? Now we are using so many booleans all over the place, which would be harder to maintain.

Could we use for example a new `MediaProviderCapability` class with an enum of capabilties (e.g. `MediaProviderCapability.PLAYLIST_EDIT`. Then as a property of each provider we can just use:
`_supported_capabilities: List[MediaProviderCapability]`

File: music_assistant/music_providers/spotify.py [hunk 1]
```diff
@@ -45,20 +45,31 @@ class SpotifyProvider(MusicProvider):
...
+        return (
+            MusicProviderFeature.LIBRARY_ARTISTS,
+            MusicProviderFeature.LIBRARY_ALBUMS,
+            MusicProviderFeature.LIBRARY_TRACKS,
+            MusicProviderFeature.LIBRARY_PLAYLISTS,
+            MusicProviderFeature.LIBRARY_RADIOS,
+            MusicProviderFeature.LIBRARY_ARTISTS_EDIT,
+            MusicProviderFeature.LIBRARY_ALBUMS_EDIT,
+            MusicProviderFeature.LIBRARY_PLAYLISTS_EDIT,
+            MusicProviderFeature.LIBRARY_RADIOS_EDIT,
```
Comment (MarvinSchenkel) [inline music_assistant/music_providers/spotify.py:67]:
```suggestion
            MusicProviderFeature.LIBRARY_RADIOS_EDIT,
            MusicProviderFeature.LIBRARY_TRACKS_EDIT,
```

File: music_assistant/music_providers/qobuz.py [hunk 2]
```diff
@@ -39,15 +39,27 @@ class QobuzProvider(MusicProvider):
...
+        return (
+            MusicProviderFeature.LIBRARY_ARTISTS,
+            MusicProviderFeature.LIBRARY_ALBUMS,
+            MusicProviderFeature.LIBRARY_TRACKS,
+            MusicProviderFeature.LIBRARY_PLAYLISTS,
+            MusicProviderFeature.LIBRARY_RADIOS,
+            MusicProviderFeature.LIBRARY_ARTISTS_EDIT,
+            MusicProviderFeature.LIBRARY_ALBUMS_EDIT,
+            MusicProviderFeature.LIBRARY_PLAYLISTS_EDIT,
+            MusicProviderFeature.LIBRARY_RADIOS_EDIT,
```
Comment (MarvinSchenkel) [inline music_assistant/music_providers/qobuz.py:57]:
```suggestion
            MusicProviderFeature.LIBRARY_RADIOS_EDIT,
            MusicProviderFeature.LIBRARY_TRACKS_EDIT,
```

File: music_assistant/music_providers/filesystem.py [hunk 3]
```diff
@@ -92,12 +92,24 @@ class FileSystemProvider(MusicProvider):
...
+        return (
+            MusicProviderFeature.LIBRARY_ARTISTS,
+            MusicProviderFeature.LIBRARY_ALBUMS,
+            MusicProviderFeature.LIBRARY_TRACKS,
+            MusicProviderFeature.LIBRARY_PLAYLISTS,
+            MusicProviderFeature.LIBRARY_RADIOS,
+            MusicProviderFeature.LIBRARY_ARTISTS_EDIT,
+            MusicProviderFeature.LIBRARY_ALBUMS_EDIT,
+            MusicProviderFeature.LIBRARY_PLAYLISTS_EDIT,
+            MusicProviderFeature.LIBRARY_RADIOS_EDIT,
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply music_assistant/music_providers/filesystem.py:108]: ik heb het in elk geval consequent gecopy paste :-)

File: music_assistant/models/music_provider.py [hunk 4]
```diff
@@ -64,14 +62,22 @@ def available(self) -> bool:
...
     @property
-    def supports_browse(self) -> bool:
-        """Return boolean if this provider supports browsing."""
-        return self._attr_supports_browse
-
-    @property
-    def supported_mediatypes(self) -> List[MediaType]:
-        """Return MediaTypes the provider supports."""
-        return self._attr_supported_mediatypes
+    def supported_features(self) -> Tuple[MusicProviderFeature]:
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply music_assistant/models/music_provider.py:65]: not sure, most providers support this but we could also just do an empty tuple...

**Pull request merged**